File: src/game/tools/rpc/output/wrap/StringKVWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class StringKVWrap : IWrapper
	{	
		public string Key; // map的key
		public string Value; // map的value
		
		public void Decode(Packet pck)
		{	
			Key = pck.GetString(); 
			Value = pck.GetString(); 
		}
		
        public void Encode(Packet pck)
        {	
        	pck.PutString(Key); 
        	pck.PutString(Value); 
        }
	}
}

File: src/game/tools/rpc/output/wrap/StringKVWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type StringKVWrap struct {
	
	Key string // map的key
	Value string // map的value
	rpc.Wrapper
}

func (w *StringKVWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.Key = pck.PopString(); 
	w.Value = pck.PopString(); 
	return w
}

func (w *StringKVWrap)Encode(pck *rpc.Packet) {
	
	pck.PutString(w.Key) 
	pck.PutString(w.Value) 
}

